src/Microsoft.Framework.PackageManager/Publish/PublishManager.cs

```diff
@@ -112,7 +113,23 @@ public bool Publish()
 
             foreach (var runtime in _options.Runtimes)
             {
-                var frameworkName = DependencyContext.GetFrameworkNameForRuntime(Path.GetFileName(runtime));
+                // Calculate the runtime name by taking the last path segment (since it could be a path),
+                // but first strip off any trailing '\' or '/' in case the path provided was something like
+                //  "C:\Foo\Bar\dnx-clr-win-x64...\"
+                var runtimeName = Path.GetFileName(runtime.TrimEnd(Path.DirectorySeparatorChar));
```

Comment:
I would suggest `new DirectoryInfo(runtime).Name` instead of this. It is possible that we are on Windows but a user provides a path with trailing forward slash.